src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -312,6 +323,60 @@ class EthService(
     }
   }
 
+  def getBalance(req: GetBalanceRequest): ServiceResponse[GetBalanceResponse] = {
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        GetBalanceResponse(account.balance)
+      }
+    }
+  }
+
+  def getStorageAt(req: GetStorageAtRequest): ServiceResponse[GetStorageAtResponse] = {
+    import io.iohk.ethereum.rlp.UInt256RLPImplicits._
+
+    val byteArrayUInt256Serializer = new ByteArrayEncoder[UInt256] {
+      override def toBytes(input: UInt256): Array[Byte] = input.bytes.toArray[Byte]
+    }
+
+    val rlpUInt256Serializer = new ByteArraySerializable[UInt256] {
```

Comment:
Mpt will become a module or a part of another one (I'm actually towards creating a a separate one as Mpt is a data structure itself and can be used in other contexts) 

That being said, I think that your solution, @rtkaczyk, it's the best one.